plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -379,6 +383,21 @@ public void flushPartitionCache(String schemaName, String tableName, List<String
         invalidatePartitionCache(schemaName, tableName, partitionNameToCheck -> partitionNameToCheck.map(value -> value.equals(providedPartitionName)).orElse(false));
     }
 
+    private AtomicReference<PartitionStatistics> refreshTableStatistics(HiveTableName tableName, AtomicReference<PartitionStatistics> currentValueHolder)
+    {
+        PartitionStatistics currentValue = currentValueHolder.get();
+        if (currentValue == null) {
+            // do not refresh empty value
+            return currentValueHolder;
+        }
+        PartitionStatistics reloaded = getTable(tableName.getDatabaseName(), tableName.getTableName())
+                .map(table -> table.withSelectDataColumnsOnly(currentValue.getColumnStatistics().keySet()))
+                .map(delegate::getTableStatistics)
+                .orElseThrow();
+        currentValueHolder.accumulateAndGet(reloaded, (latestValue, newValue) -> mergePartitionColumnStatistics(latestValue, reloaded));
+        return currentValueHolder;
```

Comment:
nit: `CachingHiveMetastore::mergePartitionColumnStatistics`

more importantly though, i would expect the fresh to produce a new value, instead of carrying over previously loaded statistics. doesn't refresh insert a new value into the cache?
i wouldn't be surprised to see here
```
return new AtomicReference<>(reloaded);
```

(without modifying `currentValue`)

Reply:
reloader loads a new, fresh cache value. usually, the cached values are immutable. the fact that our particular cache uses mutable, incrementally refined cache values is a special case

i actually don't know how guava cache handles reload returning same value as the original. does it consider this a succesful load? 

i.e. do you see a reason why we should return previous AtomicReference?
to me it's an unnecessary complication making thinking about this cache more difficult (the fresh value is not guaranteed to be fresh)